### DISCUSSION TITLE
Github Action Tutorial Feedback

### DISCUSSION THREAD

**Discussion by Oceanestars** [skills/.github#7] [General]
Hi!
Just finished the GitHub action tutorial which was my first ever exposure to GitHub actions(and my first course through GitHub skills)!
Workflow seems to closely relate to pipeline( is there a difference between the two?).

I have some feedback, hopefully, it's helpful :)(take however much you want)!

**Pros:**
- The number of resources provided was greatly appreciated.
-  The amount of info and length of this was very easy.
- The format was very engaging.

**Cons:**
- I was left with a few questions
- Some steps were very high level. I know it's an introduction but I wish we would have gone over the things that were already given to me.

**Suggestion:**
- A quick walkthrough of the actions provided with the repo(Step 0 - 5)(beyond what's in the read me, for example, I went through the whole tutorial and I kept receiving the emails about the failing workflow but didn't fully understand and this sentence "and an action will automatically close this step and open the next one." didn't mean much until closer to the end. I recommend maybe pointing to "Step 1, Create a workflow" in the actions tab and point how the action closed the step inside the yaml file. (No need to do that for every step but having a little bit more context of what's happening at all times would be cool.

- A longer walkthrough of the actions tab. I found myself wondering why so many steps were failing (like [Check emoji shortcode](https://github.com/Oceanestars/Github-Actions-Tutorial/runs/6767707061?check_suite_focus=true#logs))

- Optionally, It could be cool if you made the course taker create a step 6 workflow and get them halfway to a more complicated way of maybe scheduling actions.

Overall, this was very enjoyable and I appreciate all the efforts put into this course!

**Comment by gmondello** [comment]
my experience is similar to @Oceanestars. Really enjoy the interactive nature of these tutorials. I think there may be a syntactical error in Line 7 of the yml file... and hopefully someone smarter than me can fix it :)

**Reply by dannymonareng** [reply]
I agree that most things are intimidating before you start, however after the first step you gain confidence and begin to enjoy the results.

**Comment by SuperRoach** [comment]
I think somethings wrong with the syntax - Like you're grepping for [^ -~] inside the readme.md file, adding a variety of emojis would error out - I copy and pasted a few 😐 like emoji which are not short code and think that was wrong compared to a :code: style one. I would suggest having another part that comments on say how large the file is, to prove something has changed as well.

**Comment by volksec** [comment]
It was a great experience!

**Comment by damitr** [comment]
I am not able to complete **Step 2**. substep **5. Commit your changes.**  
Though I am able to edit the _config.yml file, I am not able to commit changes. The commit changes button is not clickable and is grayed out. What could be the issue?

**Comment by gmondello** [comment]
can you add a screenshot here to make it easier to diagnose?

**Reply by damitr** [reply]
![Screenshot 2022-06-09 at 18 51 41](https://user-images.githubusercontent.com/5547691/172857262-e312cfc3-6534-4359-b823-034a49c27eae.jpg)
![Screenshot 2022-06-09 at 18 51 35](https://user-images.githubusercontent.com/5547691/172857266-6e6d4b6a-c942-4abe-b789-c02d412f44f8.jpg)

Here is the editor, as you can see the "Commit Changes" button is not enabled.

**Comment by yhakamay** [comment]
I enjoyed it, though I'm a kind of intermediate user. GitHub Actions for collapsing instructions were significant; I didn't even know that 😳

**Reply by xiganbana** [reply]
Like what we continue to learn every day ^^, otherwise very good course for beginners.

**Comment by pausad** [comment]
where is the training site ?

**Reply by Oceanestars** [reply]
Are you talking about this https://github.com/skills?

**Comment by NigelRowlan** [comment]
Step 2 - I cannot get this to work at all.  In ci.yml -

Run npm install remark-cli remark-preset-lint-consistent

added 179 packages, and audited 180 packages in 6s
89 packages are looking for funding
  run `npm fund` for details
found 0 vulnerabilities
README.md
  115:1-115:63  warning  Emphasis should use `_` as a marker  emphasis-marker  remark-lint
⚠ 1 warning

Error: Process completed with exit code 1.


And in the instructions, the english does not make sense -

"Update the code in the ci branch to get the test to pass. You need to look something like this:
_underscore_"

This is completely clueless and I'm giving up.

**Reply by mrpatricko** [reply]
I came here to make the same comment. I can't seem to fix the error either.

Can't we just see a screenshot of what we need to type in order to fix the error? "underscore" is so vague.

----
EDIT:

For anyone else that got stuck on step 2, you just need to goto line 115 in the README.md file and add an underscore to both ends of that line. (also delete the asterisk).

![step-2](https://user-images.githubusercontent.com/2233378/180577837-7e6b30df-a502-4cde-8f60-0348769bc48a.png)

**Reply by Mbuyamba** [reply]
Thank you! Very helpful.

**Reply by Data-Engineer-Joe** [reply]
i can feel you! I was stuck for 2hours and had no clue! _underscore_ didn't help anything. even step 1 was so unclear where to put things, even to find ci.yml. oh wait it's somewhere in some folders... uf

**Comment by ghost** [comment]
I really liked it nd It really helped me .

**Comment by Arzumtn** [comment]
Response

**Comment by Arzumtn** [comment]
> Hi! Just finished the GitHub action tutorial which was my first ever exposure to GitHub actions(and my first course through GitHub skills)! Workflow seems to closely relate to pipeline( is there a difference between the two?).
> 
> I have some feedback, hopefully, it's helpful :)(take however much you want)!
> 
> **Pros:**
> 
> * The number of resources provided was greatly appreciated.
> * The amount of info and length of this was very easy.
> * The format was very engaging.
> 
> **Cons:**
> 
> * I was left with a few questions
> * Some steps were very high level. I know it's an introduction but I wish we would have gone over the things that were already given to me.
> 
> **Suggestion:**
> 
> * A quick walkthrough of the actions provided with the repo(Step 0 - 5)(beyond what's in the read me, for example, I went through the whole tutorial and I kept receiving the emails about the failing workflow but didn't fully understand and this sentence "and an action will automatically close this step and open the next one." didn't mean much until closer to the end. I recommend maybe pointing to "Step 1, Create a workflow" in the actions tab and point how the action closed the step inside the yaml file. (No need to do that for every step but having a little bit more context of what's happening at all times would be cool.
> * A longer walkthrough of the actions tab. I found myself wondering why so many steps were failing (like [Check emoji shortcode](https://github.com/Oceanestars/Github-Actions-Tutorial/runs/6767707061?check_suite_focus=true#logs))
> * Optionally, It could be cool if you made the course taker create a step 6 workflow and get them halfway to a more complicated way of maybe scheduling actions.
> 
> Overall, this was very enjoyable and I appreciate all the efforts put into this course!

**Comment by jacobjauregui** [comment]
Wow!!! It was a challenge for me. But thanks to this I managed to learn much more about different really useful programming tools !!! :+1:

**Comment by JirawatChanthamail** [comment]
M

**Comment by anhLd184032** [comment]
Good, study easily

**Comment by velazmatt** [comment]
genial

**Comment by epellejero** [comment]
Excelente !

**Comment by teboho** [comment]
Was a fun experience to go through! <3

**Reply by avilanicasionan92** [reply]
-

**Comment by avilanicasionan92** [comment]
Oli

**Comment by Irfanath** [comment]
I really like it! , first time i understand the purpose of Github and process. reaaly interesting!

**Comment by Techlearner08** [comment]
It's nice and clear, just need more practise to get familiar with each functions
